docs/docs/reference/dependent-function-types.md

```diff
@@ -0,0 +1,48 @@
+---
+layout: doc-page
+title: "Dependent Function Types"
+---
+
+A dependent function type describes functions where the result type may depend
+on the function's parameter values. Example:
+
+    class Keyed { type Key; key: Key }
```

Comment:
`Keyed ` and `Key` are a bit too easy to confuse when following the example. I would maybe replace `Keyed` by `Entry`.

Reply:
OK, let's do this.